src/common/contexts/ThemeProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useContext } from 'react';
+
+type Theme = 'light' | 'dark';
+type ThemeContext = { theme: Theme; toggleTheme: () => void };
+type ThemeProviderProps = {
+  children: React.ReactNode;
+};
+
+export const ThemeContext = React.createContext<ThemeContext>(
+  {} as ThemeContext,
+);
+
+export const ThemeProvider = ({ children }: ThemeProviderProps) => {
+  const [theme, setTheme] = useState<Theme>('light');
+  const toggleTheme = () => {
+    setTheme(theme === 'light' ? 'dark' : 'light');
+  };
+
+  const color = theme === 'light' ? '#000' : '#fff';
+  const backgroundColor = theme === 'light' ? '#fff' : '#000';
+
+  if (typeof document !== 'undefined') {
```

Comment:
Yea, this is not good, read this, this and this.

https://styled-components.com/docs/advanced#theming

https://styled-components.com/docs/advanced#via-usetheme-custom-hook

https://styled-components.com/docs/api#typescript

Reply:
made some updates around this, not sure if it's ok